src/os/xdg.zig

```diff
@@ -30,6 +30,13 @@ pub fn config(alloc: Allocator, opts: Options) ![]u8 {
 
 /// Get the XDG cache directory. The returned value is allocated.
 pub fn cache(alloc: Allocator, opts: Options) ![]u8 {
+    if (posix.getenv("XDG_CACHE_HOME")) |env| {
+        return try std.fs.path.join(alloc, &[_][]const u8{
+            env,
+            opts.subdir orelse "",
+        });
+    }
```

Comment:
Is this necessary? Doesn't `dir` (called below) do exactly this?

Reply:
As a Zig beginner, I over-engineered this. I thought it would be more efficient and secure, but in fact, the `dir()` has already handled this.